src/repo.rs

```diff
@@ -803,6 +814,7 @@ enum WakeSignal {
     PendingCloseSink(RepoId),
     Storage(DocumentId),
     StorageList,
+    ShareDecision(RepoId),
```

Comment:
Here you could add the `DocumentId`, to lookup a `DocumentInfo`, and then lookup the authorization in `HashMap<RepoId, SyncAuthorization>`